service/resource/clusterrolebindingv2/resource.go

```diff
@@ -0,0 +1,107 @@
+package clusterrolebindingv2
+
+import (
+	"reflect"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/framework"
+	apiv1 "k8s.io/api/rbac/v1beta1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	// Name is the identifier of the resource.
+	Name = "clusterrolebindingv2"
+
+	VersionBundleVersionAnnotation = "giantswarm.io/version-bundle-version"
+)
+
+// Config represents the configuration used to create a new config map resource.
+type Config struct {
+	// Dependencies.
+	K8sClient kubernetes.Interface
+	Logger    micrologger.Logger
+}
+
+// DefaultConfig provides a default configuration to create a new config map
+// resource by best effort.
+func DefaultConfig() Config {
+	return Config{
+		// Dependencies.
+		K8sClient: nil,
+		Logger:    nil,
+	}
+}
+
+// Resource implements the config map resource.
+type Resource struct {
+	// Dependencies.
+	k8sClient kubernetes.Interface
+	logger    micrologger.Logger
+}
+
+// New creates a new configured config map resource.
+func New(config Config) (*Resource, error) {
+	// Dependencies.
+	if config.K8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.K8sClient must not be empty")
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Logger must not be empty")
+	}
+
+	newService := &Resource{
+		// Dependencies.
+		k8sClient: config.K8sClient,
+		logger: config.Logger.With(
+			"resource", Name,
+		),
+	}
+
+	return newService, nil
+}
+
+func (r *Resource) Name() string {
+	return Name
+}
+
+func (r *Resource) Underlying() framework.Resource {
+	return r
+}
+
+func containsClusterRoleBinding(list []*apiv1.ClusterRoleBinding, item *apiv1.ClusterRoleBinding) bool {
+	_, err := getClusterRoleBindingByName(list, item.Name)
+	if err != nil {
+		return false
```

Comment:
An error can indicate any kind of unexpected behaviour. What we want here is to make sure the role binding cannot be found. This is the only kind of error we expect and only in this case we should return false. Tomorrow I change something and am not aware of all the implementation details. Then I can screw the implementation dangerously and nobody notices until the customer complains. I would suggest doing something like this. 

```
if IsRoleBindingNotFound(err) {
  return false, nil
} else if err != nil {
  return false, microerror.Mask(err)
}
```